### ISSUE TITLE
calling detach on a numpy array in art.estimators.regression.pytorch

### ISSUE THREAD

**Issue by annapmeyer** [Trusted-AI/adversarial-robustness-toolbox#1823]
**Describe the bug**
Lines 270-272 in art.estimators.regression.pytorch.py are:
```
output = output.detach().cpu().numpy().astype(np.float32)
if len(output.shape) == 1:
    output = np.expand_dims(output.detach().cpu().numpy(), axis=1).astype(np.float32)
```

Line 270 sets output to be a numpy array, but then line 272 calls detach() on output, resulting in a type error since detach() is a method for tensors, not numpy arrays.

**To Reproduce**
Call ```predict()``` on a ```PyTorchRegressor``` object that returns a 1-D  output.

**Expected behavior**
the code should not cause a bug and crash.

**Screenshots**
The exact error is "'numpy.ndarray' object has no attribute 'detach'"

**System information (please complete the following information):**
 - OS: Mac Monterey 12.4
 - Python version: 3.8.10
 - ART version or commit number: main branch as of 8/18/22
 - TensorFlow / Keras / PyTorch / MXNet version: PyTorch version 1.12.1

**Comment by beat-buesser** [comment]
Hi @annapmeyer Thank you very much for discovering and fixing this bug!
